File: api/Entities/Admin/Employee.cs
using System.ComponentModel.DataAnnotations;
using api.Entities.HR;
using AutoMapper.Configuration.Conventions;

namespace api.Entities.Admin
{
    public class Employee: BaseEntity
    {
        public int AppUserId { get; set; }
        public string Gender { get; set; }
        [Required]
        public string FirstName { get; set; }
        public string SecondName { get; set; }
        public string FamilyName { get; set; }
        [Required]         
        public string KnownAs { get; set; }
        public string DisplayName {get; set;}
        public string UserName { get; set; }
        [Required]
        public string Position { get; set; }
        public string Qualification {get; set;}
        [Required]
        public DateTime DateOfBirth { get; set; }
        public string PlaceOfBirth { get; set; }
        public string AadharNo { get; set; }
       [EmailAddress]
        public string Email {get; set;}
        public string PhoneNo { get; set; }
        public string Phone2 { get; set; }
        public DateTime DateOfJoining {get; set;}
        public string Department { get; set; }
        public string Remarks { get; set; }        
        public string Status { get; set; } = "Employed";
        public string Address { get; set; }
        public string Address2 {get; set;}
        public string City { get; set;} 
        public ICollection<HRSkill> HRSkills {get; set;}
        public ICollection<EmployeeOtherSkill> EmployeeOtherSkills{get; set;}
        public ICollection<EmployeeAttachment> EmployeeAttachments {get; set;}  
    }
}

File: client/src/app/_models/admin/employee.ts
import { IHRSkill } from "./hrSkill";
import { IEmployeeOtherSkill } from "./employeeOtherSkill";
import { IEmployeeAttachment } from "./employeeAttachment";

export interface IEmployee {
     id: number;
     gender: string;
     firstName: string;
     secondName: string;
     familyName: string;
     knownAs: string;
     displayName: string;
     address: string;
     address2: string;
     city: string;
     //pin: string;
     //state: string;
     
     email: string;
     phoneNo: string;
     phone2: string;
     dateOfBirth: Date;
     dateOfJoining: Date;     

     placeOfBirth: string;
     department: string;
     aadharNo: string;
     introduction: string;
     position: string;
     appUserId: number;
     qualification: string;
    
     userName: string;
     //remarks: string;
     status: string;

     hrSkills: IHRSkill[];
     employeeOtherSkills: IEmployeeOtherSkill[];
     employeeAttachments: IEmployeeAttachment[];
}    

export class Employee implements IEmployee {
     id= 0;
     gender= 'M';
     firstName= '';
     secondName= '';
     familyName= '';
     knownAs= '';
     displayName= '';
     address='';
     address2= '';
     city= '';
     //pin= '';
     //state= '';
     
     email= '';
     phoneNo = '';
     phone2 = '';
     dateOfBirth = new Date;
     dateOfJoining= new Date;     

     placeOfBirth= '';
     department= '';
     aadharNo= '';
     introduction= '';
     position= '';
     appUserId= 0;
     qualification = '';
     userName= '';
     //remarks='';
     status='';

     hrSkills: IHRSkill[]=[];
     employeeOtherSkills: IEmployeeOtherSkill[]=[];
     employeeAttachments: IEmployeeAttachment[]=[];
}    
